src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -644,6 +689,24 @@ private String directiveString(GraphQLDirective directive) {
         return sb.toString();
     }
 
+    private boolean isDeprecatedDirective(GraphQLDirective directive) {
+        return directive.getName().equals(DeprecatedDirective.getName());
+    }
+
+    private boolean hasDeprecatedDirective(List<GraphQLDirective> directives) {
+        return directives.stream()
+                .filter(this::isDeprecatedDirective)
+                .count() == 1;
```

Comment:
Considering this is a list and not a set, can we ever have `count > 1`, in which case should this be `count() >= 1` to work for all these scenarios? Or have we built enough invariants for this class that this shouldn't happen?

Reply:
Hmm good point - the rest of the code base puts `List<GraphQLDirective>` everywhere and hence its influenced this code as well

I think other parts of the code stop multiple named directives